server/container_create.go

```diff
@@ -702,14 +702,25 @@ func (s *Server) CreateContainer(ctx context.Context, req *pb.CreateContainerReq
 		}
 	}()
 
-	if err = s.Runtime().CreateContainer(container, sb.CgroupParent()); err != nil {
-		return nil, err
-	}
-
 	s.addContainer(container)
+	defer func() {
+		if err != nil {
+			s.removeContainer(container)
+		}
+	}()
 
 	if err = s.CtrIDIndex().Add(containerID); err != nil {
-		s.removeContainer(container)
+		return nil, err
+	}
+	defer func() {
+		if err != nil {
+			if err2 := s.CtrIDIndex().Delete(containerID); err2 != nil {
+				logrus.Warnf("couldn't delete ctr id %s from idIndex", containerID)
```

Comment:
Add err2 string to message

Reply:
we're not doing that in other cases because I believe the error is not saying anything useful (just something about the trunc/patricia index).